structuralcodes/codes/ec2_2023/_section5_materials.py

```diff
@@ -0,0 +1,1118 @@
+    if A_exponent > 0.82:
+        raise ValueError(f'A_exponent={A_exponent} cannot be less than 0.82')
+
+    return math.pow(35 / fck, A_exponent)
+
+
+def phi_50y_t0(
+    t0: float, atm_conditions: str, _hn: float, concrete_class: str
+) -> float:
+    """Computes the creep coefficient of plain concrete at 50 years
+    of loading. Interpolation is lineal between values
+
+    FprEN 1992-1-1, Table 5.2
+
+    Args:
+        t0 (float): age at loading [days]
+        atm_conditions (str): 'dry' or 'humid'
+        _hn (float): the notional size in mm
+        concrete_class (str): 'CS', 'CN' or 'CR'
+
+    Returns:
+        float: the creep coefficient
+
+    Raises:
+        ValueError: if t0 is less than 0
+        ValueError: if atm_conditions is not 'dry' or 'humid'
+        ValueError: if _hn is less than 100 or larger than 1000
+        ValueError: if concrete_class is not 'CS', 'CN' or 'CR'
+        ValueError: if combination of t0 and _hn is out of scope
+    """
+    if t0 < 0:
+        raise ValueError(f't0={t0} cannot be less than 0')
+
+    atm_conditions = atm_conditions.lower().strip()
+    if atm_conditions not in ('dry', 'humid'):
+        raise ValueError(
+            f'atm_conditions={atm_conditions} must be "dry" or "humid"'
+        )
+
+    if _hn < 100:
+        raise ValueError(f'_hn={_hn} must be larger or equal than 100')
+    if _hn > 1000:
+        raise ValueError(f'_hn={_hn} must be less or equal than 1000')
+
+    concrete_class = concrete_class.upper().strip()
+    if concrete_class not in ('CS', 'CN', 'CR'):
+        raise ValueError(
+            f'concrete_class={concrete_class} must be "CS", "CN" or "CR"'
+        )
+
+    if concrete_class == 'CS':
+        _t = (3, 10, 32, 91, 365)
+    elif concrete_class == 'CN':
+        _t = (1, 7, 28, 91, 365)
+    elif concrete_class == 'CR':
+        _t = (1, 3, 23, 91, 365)
+
+    h_v = (100, 200, 500, 1000)
+
+    if atm_conditions == 'dry':
+        values = (
+            4.2,
+            3.8,
+            3.4,
+            3.1,
+            3.1,
+            2.8,
+            2.5,
+            2.3,
+            2.4,
+            2.2,
+            1.9,
+            1.8,
+            1.9,
+            1.7,
+            1.5,
+            1.4,
+            1.4,
+            1.3,
+            1.1,
+            1.0,
+        )
+    elif atm_conditions == 'humid':
+        values = (
+            3.0,
+            2.8,
+            2.6,
+            2.5,
+            2.2,
+            2.1,
+            2.0,
+            1.9,
+            1.7,
+            1.6,
+            1.6,
+            1.5,
+            1.4,
+            1.3,
+            1.2,
+            1.2,
+            1.0,
+            0.9,
+            0.9,
+            0.8,
+        )
+
+    grid = np.array(np.meshgrid(_t, h_v)).T.reshape(-1, 2)
+    p = (t0, _hn)
+
+    interp = scipy.interpolate.griddata(grid, values, p, method='linear')
```

Comment:
![image](https://github.com/fib-international/structuralcodes/assets/80036012/4d189e03-34cc-4171-b7a3-e4e7bc0f2248)

If you take a look to the last paramitrized test, when `concrete_class=CS` and `t0=1`day it will raise a ValueError. If you consider that we need to extrapolate the data, I can change it.